java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -108,16 +108,29 @@
    */
   static int findColumns(SchemaEvolution evolution,
                          String columnName) {
-    TypeDescription readerSchema = evolution.getReaderBaseSchema();
+
+    TypeDescription result = evolution.getFileType(findColumns(evolution.getReaderBaseSchema(),columnName));
+    return result == null ? -1 : result.getId();
```

Comment:
Simple column names are getting passed here. 
The case where the nested structure could also have the same column name as the outer node is not taken care by the original ORC code itself. I think you might have to change (columnName) what is getting passed to this function to some cannonical representation. Do check with the community.